### PR TITLE
Add Babel Pipeline, Better ESLint

### PR DESCRIPTION
**Pull request by damassi**
This PR adds Babel to the flow, along with some common packages as taken from Force:

- babel-cli
- babel-plugin-inline-react-svg
- babel-plugin-module-resolver 
- babel-plugin-rewire
- babel-plugin-styled-components
- babel-plugin-transform-class-properties
- babel-plugin-transform-runtime
- babel-preset-es2015
- babel-preset-react
- babel-preset-stage-3

Notable items from above:
- Updates ESLint setup to follow what we have on Force, with `standard` and `standard-react` being the primary engines. 
- In React code, `.svg` file imports are automatically converted into components that you can use like so:  `<Icon />` 
- Modules written in JS can take advantage of absolute path imports -- e.g., `import baz from 'api/foo/bar/baz'`
- Similar to Force, rewrite for test code
- I know most component work is taking place in `reaction`, but I want to setup server-side react rendering and this, similar to force, is needed to pluck those styles out

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+require('babel-core/register')
+require('coffee-script/register')
+require('./app')
```
Comment (damassi) [inline index.js]: Created this boot loader so that from the point of `./app` onwards we can write ES6.

File: index.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+if (process.env.NODE_ENV === 'test') {
+  env('./.env.test')
+
+  // Other envs: development, staging, production
+} else {
+  env('./.env')
+}
+
+require('./boot')
```
Comment (damassi) [inline index.js]: Created this boot loader so that from the point of ./app onwards we can write ES6.

File: boot.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      process.send('listening')
+    }
+  })
+})
+
+// Crash if we can't get/refresh an xapp token
+artsyXapp.on('error', (error) => {
+  console.warn(error)
+  process.exit(1)
+})
```
Comment (kanaabe) [inline boot.js]: 🙌

**Comment by kanaabe** [comment]
👏 This is so awesome!

**Comment by damassi** [comment]
😄

**Pull request merged**